massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -1095,7 +1095,19 @@ impl ProtocolWorker {
         let mut seen_ops = vec![];
         let mut new_operations = Map::with_capacity_and_hasher(length, BuildMap::default());
         let mut received_ids = Map::with_capacity_and_hasher(length, BuildMap::default());
-        for (idx, operation) in operations.into_iter().enumerate() {
+
+        // Ensure we have a Vec<Option<Vec<u8>>> to zip with the Operations.
+        let serialized = if let Some(serialized) = serialized {
+            serialized.into_iter().map(Some).collect()
+        } else {
+            vec![None; length]
```

Comment:
I think we could directly serialize it here instead of returning None. It will simplify the code below